tests/unit/cluster/faster-failover.tcl

```diff
@@ -0,0 +1,164 @@
+        } else {
+            fail "No failover detected"
+        }
+
+        # case 1: R0 and R1 down in the same time, R0 have a better failed primary rank, R10 is
+        # the best ranked replica in the first place, if so, there is no delay.
+        if {$R0_failed_primary_rank == 0} {
+            if {[count_log_message -10 "This is the best ranked replica and can initiate the election immediately"] != 0} {
+                verify_log_message -10 "*Start of election delayed for 0 milliseconds*" 0
+            }
+        }
+        # No matter what, R10 will be the best ranked replica in R0.
+        # This is the best ranked replica and can initiate the election immediately
+        # Myself become the best ranked replica, initiate the election immediately
+        verify_log_message -10 "*best ranked replica*" 0
+        wait_for_log_messages -5 {"*Successful partial resynchronization with primary*"} 0 1000 50
+
+        # case 2: R0 and R1 down in the same time, R1 have a better failed primary rank, R6 or R11
+        # will be the best ranked replica in the first place, if so, there is no delay.
+        if {$R1_failed_primary_rank == 0 && $R6_replica_rank == 0} {
+            if {[count_log_message -6 "This is the best ranked replica and can initiate the election immediately"]} {
+                verify_log_message -6 "*Start of election delayed for 0 milliseconds*" 0
+            }
+        }
+        if {$R1_failed_primary_rank == 0 && $R11_replica_rank == 0} {
+            if {[count_log_message -11 "This is the best ranked replica and can initiate the election immediately"]} {
+                verify_log_message -11 "*Start of election delayed for 0 milliseconds*" 0
+            }
+        }
+        # No matter what, there will be a best ranked replica in R1.
+        # This is the best ranked replica and can initiate the election immediately
+        # Myself become the best ranked replica, initiate the election immediately
+        if {$R6_replica_rank == 0} {
+            verify_log_message -6 "*best ranked replica*" 0
+            wait_for_log_messages -11 {"*Successful partial resynchronization with primary*"} 0 1000 50
+        } else {
+            verify_log_message -11 "*best ranked replica*" 0
+            wait_for_log_messages -6 {"*Successful partial resynchronization with primary*"} 0 1000 50
+        }
+
+        # In any case, we do not expect timeouts.
+        verify_no_log_message -5 "*Failover attempt expired*" 0
+        verify_no_log_message -10 "*Failover attempt expired*" 0
+        verify_no_log_message -6 "*Failover attempt expired*" 0
+        verify_no_log_message -11 "*Failover attempt expired*" 0
+    }
+}
+
+# This change and test is quite important, so we want to run it a few more times.
+
+start_cluster 5 7 {tags {external:skip cluster} overrides {cluster-ping-interval 1000 cluster-node-timeout 5000}} {
+    test_best_ranked_replica
+} my_slot_allocation cluster_allocate_replicas ;# start_cluster
```

Comment:
Will you remove this before merging?

If you want to keep running it multiple times, wrap the code in for loop instead of duplicating the code.

Reply:
ohh right, i change to use a loop, i indeed want to run it for a few times so that we are not messing this up. 5 times take 120 seconds in my PC, so i think that is acceptable (we can change it in the furture if we think they are slow).